File: backend/mallapi/src/main/java/org/zerock/mallapi/domain/Todo.java
package org.zerock.mallapi.domain;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Table(name = "tbl_todo")
@Getter
@ToString // toString 메서드를 자동으로 생성해서 엔티티값을 출력하기 쉽게 만들어줌
@Builder  
@AllArgsConstructor // 모든 필드를 파라미터로 받는 전체 생성자 자동생성
@NoArgsConstructor  // 파라미터 없는 기본생성자 생성
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long tno;

    private String title;

    private String writer;

    private boolean complete;

    private LocalDate dueDate;

    public void changeTitle(String title){
        this.title = title;
    }
    public void changeComplete(boolean complete){
        this.complete = complete;
    }
    public void changeDueDate(LocalDate dueDate){
        this.dueDate = dueDate;
    }
}

File: frontend/src/types/todo.ts
// src/types/todo.ts

export type TodoType = {
  tno: number;
  title: string;
  writer: string;
  dueDate: string | null;
  complete: boolean;
};

export type TodoAddType = {
  title: string;
  writer: string;
  dueDate: string;
};

export type TodoModifyType = {
  tno: number;
  title: string;
  writer: string;
  dueDate: string;
  complete: boolean;
};

export type GetListParams = {
  page: number;
  size: number;
};
